File: typescript/25-rot13.ts
// Cifrado ROT13 — Fácil · Strings
// Rota letras dentro de A-Z/a-z y conserva los demás caracteres.
// Fuente: https://coding-challenges.dev/problems/typescript-cifrar-rot13

export function rot13(text: string): string {
  return text.replace(/[a-z]/gi, (char) => {
    const base = char <= "Z" ? 65 : 97;
    return String.fromCharCode((char.charCodeAt(0) - base + 13) % 26 + base);
  });
}

File: python/25_rot13.py
"""Cifrado ROT13 — Fácil · Strings

Aplica ROT13 a letras mayúsculas y minúsculas; los demás caracteres quedan
sin cambios.

Fuente: https://coding-challenges.dev/problems/python-cifrar-rot13
"""


def rot13(text: str) -> str:
    result = []
    for char in text:
        if "a" <= char <= "z":
            result.append(chr((ord(char) - ord("a") + 13) % 26 + ord("a")))
        elif "A" <= char <= "Z":
            result.append(chr((ord(char) - ord("A") + 13) % 26 + ord("A")))
        else:
            result.append(char)
    return "".join(result)
